project/src/main/java/com/google/autograder/servlets/QuestionServlet.java

```diff
@@ -0,0 +1,31 @@
+package com.google.autograder.servlets;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Arrays;
+import com.google.autograder.data.Database;
+import com.google.autograder.data.Question;
+import com.google.gson.Gson;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+
+@WebServlet("/question")
+public final class QuestionServlet extends HttpServlet {
+
+  private Database d = new Database();
```

Comment:
I noticed that all of your new servlets have their own Database instance. I think based on the way your Database class is structured this means you're creating a new separate Datastore for every servlet.

Reply:
Yeah I noticed that as well, but I thought it was just connecting to the database, not creating a new one. Is there another way that you recommend doing that?

Reply:
Check out this: https://github.com/googleinterns/step25-2020/blob/d71617c5cc84e33727d1421f99befbd40636c412/project/src/main/java/com/google/autograder/data/Database.java

I made most of the Database functionality static, so that all the servlets talk to the same Datastore. It also loops in some User Service functionality so you can grab user info and class info to cross reference from the same space.

Reply:
Note: the `Datastore` class you're using here is only a reference to a client to talk to Datastore, not the actual database itself. If you have something like:
```
DatastoreService d1 = DatastoreServiceFactory.getDatastoreService();
DatastoreService d2 = DatastoreServiceFactory.getDatastoreService();
```
both d1 and d2 will talk to the same Datastore and will work with the same set of data. There's other reasons for only using a single DatastoreService across your servlets, for example if creating a DatastoreService is expensive and should only be done once rather per request.

It is a good idea to make datastore usage consistent across your code, so conforming to what Sampson's done in that commit is probably the way to go here.